planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -634,6 +636,11 @@ void planning_scene_monitor::PlanningSceneMonitor::excludeRobotLinksFromOctree()
       if (h)
         link_shape_handles_[links[i]].push_back(std::make_pair(h, j));
     }
+    if(!warned && ((ros::Time::now() - start) > ros::Duration(30.0)))
+    {
+      ROS_WARN_STREAM_NAMED("PlanningSceneMonitor::excludeRobotLinksFromOctree()", "It is likely there are too many vertices in collision geometry");
+      warned = true;
```

Comment:
I don't like the performance implication of calling ``ros::Time::now()`` so many times. A robot like the PR2 I'm guessing has ~50 links. This warning to me belongs at the location where the meshes are first loaded into the planning scene. There we could just check the number of vertices and compare against some high bound average.